pkgs/by-name/wp/wpcleaner/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeDesktopItem,
+  copyDesktopItems,
+  makeWrapper,
+  jdk17,
+  ant,
+  stripJavaArchivesHook,
+  gettext,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wpcleaner";
+  version = "2.0.5-unstable-2025-04-25";
+  src = fetchFromGitHub {
+    owner = "WPCleaner";
+    repo = "wpcleaner";
+    rev = "7fd357cf26349658183517658139870dd45eaedc";
+    hash = "sha256-iaAP/5Z+ghvMAn4ke7lhRqKov/3jXr0LMwbPDZ052j0=";
+  };
+
+  wpcleanerJar = "$out/share/wpcleaner/WikipediaCleaner.jar";
+  clientScript = "$out/bin/wpcleaner";
+  botScript = "$out/bin/wpcleaner-bot";
+  runTaskScript = "$out/bin/wpcleaner-run-task";
+  extraJavaArgs = [
+    "-Dawt.useSystemAAFontSettings=lcd"
+    "-Xms1g"
+    "-Xmx8g"
+  ];
+
+  dontConfigure = true;
+
+  patches = [
+    ./0001-fix-script-names-and-remove-getdown.patch
+    ./0002-dont-build-javadoc.patch
+  ];
```

Comment:
Perhaps adding a comment explaining why those patches are required (could be also inferred from the filename), but also if they need to be removed at the next bump ?

Reply:
I added some comments, see https://github.com/NixOS/nixpkgs/pull/406968/files#diff-5b01801ee3ba52963a6b8316fb1d94e8f3bf63d0345838556c7dd189665e1955R36-R45. These patches are supposed to stay.